editor/store/effects/utils.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import { select, subscribe } from '@wordpress/data';
+
+/**
+ * Waits for the resolution of a selector before returning the selector's value.
+ *
+ * @param {string} namespace    Store namespace.
+ * @param {string} selectorName Selector name.
+ * @param {Array} args          Selector args.
+ *
+ * @return {Promise} Selector result.
+ */
+export function resolveSelector( namespace, selectorName, ...args ) {
```

Comment:
This utility or something equivalent will be bundled in the data module once we land the async actions work.